src/common/sha2/sha2_armv8.c

```diff
@@ -229,15 +251,68 @@ void oqs_sha2_sha224_inc_finalize_armv8(uint8_t *out, sha224ctx *state, const ui
 
 void oqs_sha2_sha256_inc_blocks_armv8(sha256ctx *state, const uint8_t *in, size_t inblocks) {
 	uint64_t bytes = load_bigendian_64(state->ctx + 32);
+	const uint8_t *new_in;
+	size_t buf_len = 64 * inblocks;
+	uint8_t *tmp_in = NULL;
+
+	/* Process any existing incremental data first */
+	if (state->data_len) {
+		tmp_in = malloc(buf_len);
+		if (tmp_in == NULL) {
+			exit(111);
```

Comment:
This appears to be a rather brutal termination. No other way to inform the user of the library as to what's happened?

Reply:
I don't find a way to properly handle this pattern. Added in #1656 : a checked_malloc to wrap the malloc check in one place. Let me know if this good enough.